common/determined_common/util.py

```diff
@@ -42,3 +44,17 @@ def get_default_master_address() -> str:
 
 def debug_mode() -> bool:
     return os.getenv("DET_DEBUG", "").lower() in ("true", "1", "yes")
+
+
+def preserve_random_state(fn: Callable) -> Callable:
+    """A decorator to run a function with a fork of the random state."""
+
+    @functools.wraps(fn)
+    def wrapped(*arg: Any, **kwarg: Any) -> Any:
```

Comment:
non-blocking: Should this be renamed to something descriptive? I'm thinking maybe wrappedSaver or similar.

Reply:
`wrappedSaver` would imply that we knew that `fn` was a saver of some sort.  This is a pretty standard format of a decorator, and the name of the decorated function is basically meaningless, in the same way that the name `fn` is basically meaningless.